commands/proposal_submission.go

```diff
@@ -267,6 +268,85 @@ func checkERC20AssetSource(s *types.AssetDetails_Erc20) Errors {
 	return errs
 }
 
+func checkUpdateAssetChanges(change *types.ProposalTerms_UpdateAsset) Errors {
+	errs := NewErrors()
+
+	if change.UpdateAsset == nil {
+		return errs.FinalAddForProperty("proposal_submission.terms.change.update_asset", ErrIsRequired)
+	}
+
+	if change.UpdateAsset.Changes == nil {
+		return errs.FinalAddForProperty("proposal_submission.terms.change.update_asset.changes", ErrIsRequired)
+	}
+
+	if len(change.UpdateAsset.Changes.Name) == 0 {
+		errs.AddForProperty("proposal_submission.terms.change.update_asset.changes.name", ErrIsRequired)
+	}
+	if len(change.UpdateAsset.Changes.Symbol) == 0 {
+		errs.AddForProperty("proposal_submission.terms.change.update_asset.changes.symbol", ErrIsRequired)
+	}
+	if change.UpdateAsset.Changes.Decimals == 0 {
+		errs.AddForProperty("proposal_submission.terms.change.update_asset.changes.decimals", ErrIsRequired)
+	}
+	if len(change.UpdateAsset.Changes.TotalSupply) == 0 {
+		errs.AddForProperty("proposal_submission.terms.change.update_asset.changes.total_supply", ErrIsRequired)
+	} else {
+		if totalSupply, ok := big.NewInt(0).SetString(change.UpdateAsset.Changes.TotalSupply, 10); !ok {
+			errs.AddForProperty("proposal_submission.terms.change.update_asset.changes.total_supply", ErrIsNotValidNumber)
+		} else if totalSupply.Cmp(big.NewInt(0)) <= 0 {
+			errs.AddForProperty("proposal_submission.terms.change.update_asset.changes.total_supply", ErrMustBePositive)
+		}
+	}
+
+	if change.UpdateAsset.Changes.Source == nil {
+		return errs.FinalAddForProperty("proposal_submission.terms.change.update_asset.changes.source", ErrIsRequired)
+	}
+
+	switch s := change.UpdateAsset.Changes.Source.(type) {
+	case *types.AssetDetailsUpdate_Erc20:
+		errs.Merge(checkERC20UpdateAssetSource(s))
+	default:
+		return errs.FinalAddForProperty("proposal_submission.terms.change.update_asset.changes.source", ErrIsNotValid)
+	}
+
+	return errs
+}
+
+func checkERC20UpdateAssetSource(s *types.AssetDetailsUpdate_Erc20) Errors {
+	errs := NewErrors()
+
+	if s.Erc20 == nil {
+		return errs.FinalAddForProperty("proposal_submission.terms.change.update_asset.changes.source.erc20", ErrIsRequired)
+	}
+
+	asset := s.Erc20
+
+	if len(asset.LifetimeLimit) == 0 {
+		errs.AddForProperty("proposal_submission.terms.change.update_asset.changes.source.erc20.lifetime_limit", ErrIsRequired)
+	} else {
+		if lifetimeLimit, ok := big.NewInt(0).SetString(asset.LifetimeLimit, 10); !ok {
+			errs.AddForProperty("proposal_submission.terms.change.update_asset.changes.source.erc20.lifetime_limit", ErrIsNotValidNumber)
+		} else {
+			if lifetimeLimit.Cmp(big.NewInt(0)) <= 0 {
+				errs.AddForProperty("proposal_submission.terms.change.update_asset.changes.source.erc20.lifetime_limit", ErrMustBePositive)
+			}
+		}
+	}
+	if len(asset.WithdrawThreshold) == 0 {
```

Comment:
In each of the error cases from this point on, we can return. If the Withdrawal threshold is empty, we don't create the big.Int, and there's nothing to check to be <= 0. This looks just a lot cleaner:

```go
if len(asset.WithdrawalThreshold) == 0 {
    return errs // add the relevant error, obviously
}
wt, ok := big.NewInt(0).SetString(...)
if !ok {
    return errs
}
if wt.Cmp(big.NewInt(0)) <= 0 {
     errs.AddForProperty()
}
return errs
```

Reply:
On this particular bloc yes, but I can't just remove it everywhere as we want it to continue the vrification (cumulative errors)